DOCSP-35974: Update one usage example (#2781)
jenssegers/laravel-mongodb: docs/usage-examples/updateOne.txt

.. _laravel-update-one-usage:

=================
Update a Document

> =================

Reviewer: I would add the `contents` for the TOC, and the tags while you're working on these pages.
```suggestion
=================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: twocols

.. facet::
   :name: genre
   :values: reference
 
.. meta::
   :keywords: <whatever you think>
```

You can update a document in a collection by chaining the ``where()``, ``first()``,
and ``update()`` methods to an Eloquent model or a query builder.

Pass a query filter to the ``where()`` method and call the ``first()`` method to match 
only one document. Then, pass the document changes to the ``update()`` method to update
one document that matches the filter.

To learn more about updating data with {+odm-short+}, see the `Updates

> <https://laravel.com/docs/6.x/eloquent#updates>`__ section of the Laravel documentation.

Reviewer 2: You can link to the latest version (currently 11.x) by removing the version from the URL.
```suggestion
<https://laravel.com/docs/eloquent#updates>`__ section of the Laravel documentation.
```

This example performs the following actions:

- Defines a query filter to match documents in which the value of the ``title`` field
  is ``'Carol'``
- Calls the ``first()`` method to retrieve the first document that matches the query filter,
  according to the sort order specified by the ``orderBy()`` method
- Calls the ``update()`` method to update the value of the ``metacritic`` field from ``98``
  to ``94``

To see the update operation code, select the :guilabel:`Controller File Code` tab. To see
the HTML code that specifies a view, select the :guilabel:`View File Code` tab.

.. important:: 

   Before running the update operation, you must add the following statement to the 
   ``Movie`` class in your ``Movie.php`` file:

   .. code-block:: php
   
      protected $fillable = array('metacritic');

   The ``$fillable`` property specifies which document fields are mass-assignable. To learn
   more about ``$fillable``, see the `Mass Assignment <https://laravel.com/docs/4.2/eloquent#mass-assignment>`__
   section of the Laravel documentation.
   
.. tabs::

   .. tab:: Controller File Code
      :tabid: find-one-controller

      .. code-block:: php

>          class MovieController

Reviewer 2: Could you extract the code example into `.php` files, so we can validate PHP syntax, fix CS and even test them.

.. _laravel-update-one-usage:

=================
Update a Document
=================

.. facet::
   :name: genre
   :values: reference

.. meta::
   :keywords: update one, modify, code example

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

You can update a document in a collection by retrieving a single document and calling
the ``update()`` method on an Eloquent model or a query builder.

Pass a query filter to the ``where()`` method, sort the matching documents, and call the

> ``first()`` method to match only one document. Then, update this matching document by passing

Reviewer 2: ```suggestion
``first()`` method to get only the first document. Then, update this matching document by passing
```
